### DISCUSSION TITLE
Can't connect to postgres container inside of github actions

### DISCUSSION THREAD

**Discussion by CarlosMaciasCusiter** [sqlalchemy/sqlalchemy#8208] [Usage Questions]
### Describe the bug

Can't connect to Database with sqlalchemy on github actions. this url worked fine when connecting to a local machine with a test DB and on a remote linux server. having issues when trying to build a CI/CD pipeline using github actions

### To Reproduce

```python
**conftest.py file**

SQLALCHEMY_DATABASE_URL = f"postgresql://{settings.database_username}:{settings.database_password}@{settings.database_hostname}:{settings.database_port}/{settings.database_name}_test"
 
engine = create_engine(SQLALCHEMY_DATABASE_URL)
```

### Error

**Output from Github Actions**

    ImportError while loading conftest '/home/runner/work/fastapi-songlist/fastapi-songlist/tests/conftest.py'.
    tests/conftest.py:13: in <module>
        engine = create_engine(SQLALCHEMY_DATABASE_URL)
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py:309: in warned
        return fn(*args, **kwargs)
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/engine/create.py:530: in create_engine
        u = _url.make_url(url)
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/engine/url.py:731: in make_url
        return _parse_rfc1738_args(name_or_url)
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/engine/url.py:789: in _parse_rfc1738_args
    postgresql://:@:/_test
        return URL.create(name, **components)
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/engine/url.py:151: in create
        cls._assert_port(port),
    /opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/sqlalchemy/engine/url.py:162: in _assert_port
        return int(port)
    E   ValueError: invalid literal for int() with base 10: ''
    Error: Process completed with exit code 4.

### Versions

- OS: ubuntu-latest
- Python: python 3.9
- SQLAlchemy: 
- Database: postgres
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): psycopg2

**Comment by CaselIT** [comment]
The error is giving you the reason:
> E   ValueError: invalid literal for int() with base 10: ''

check that your url is properly created
